File: Array/3042. Count Prefix and Suffix Pairs I/Solution.rs
impl Solution {
    pub fn count_prefix_suffix_pairs(words: Vec<String>) -> i32 {
        let mut result=0;
        for i in 0..words.len()-1{
            for j in i+1..words.len(){
                if(words[i].len()<=words[j].len()){
                    if(words[i]==words[j][0..words[i].len()]&&words[i]==words[j][words[j].len()-words[i].len()..words[j].len()]){
                        result+=1;
                    }
                }
            }
        }
        result
    }
}

File: Array/3042. Count Prefix and Suffix Pairs I/Solution.cs
public class Solution {
    public int CountPrefixSuffixPairs(string[] words) {
        int result=0;
        for(int i=0;i<words.Length-1;i++){
            for(int j=i+1;j<words.Length;j++){
                if(words[i].Length>words[j].Length){
                    continue;
                } else {
                    if(words[j].StartsWith(words[i])&&words[j].EndsWith(words[i])){
                        result++;
                    }
                }
            }
        }
        return result;
    }
}